scripts/prices.js

```diff
@@ -105,33 +108,26 @@ export class Oracle {
             }
 
             // Now we've loaded all currencies: we'll flag it and use the lower bandwidth price fetches in the future
-            this.fLoadedCurrencies = true;
+            this.#fLoadedCurrencies = true;
             return arrCurrencies;
         } catch (e) {
             console.warn('Oracle: Failed to fetch currencies!');
             console.warn(e);
             return this.getCachedCurrencies();
         }
     }
+
+    async load() {
+        while (!this.#fLoadedCurrencies) {
+            await this.getCurrencies();
+            if (!this.#fLoadedCurrencies) await sleep(5000);
+        }
+        getEventEmitter().emit('currency-loaded', this.mapCurrencies);
```

Comment:
```suggestion
        // Update any listeners for the full currency list (Settings, etc)
        getEventEmitter().emit('currency-loaded', this.mapCurrencies);
        // Update the balance to render the price instantly
        getEventEmitter().emit('balance-update');
```

Small suggestion: even after `load()` is successful after multiple failed attempts, it takes another >15s before `balance-update` is called by `global.js`, adding a delay to the currency displaying in the Dashboard, we can render both the Balances and Settings here.